### PR TITLE
[glib] Update the usage of string(TOUPPER ...)

### PR DESCRIPTION
**Pull request by NancyLi1013**
Update  `string(TOUPPER UPPER_CONFIG "${CMAKE_BUILD_TYPE}") ` as
` string("${CMAKE_BUILD_TYPE}" TOUPPER UPPER_CONFIG) `.

Related issue #10853 

Note: Feature has been passed with the following triplets:
-  x86-windows
- x64-windows

Failed on triplets:
- x64-windows-static(Only support dynamic build)
- x64-linux(Failed on master branch)

### CODE REVIEW

**Review by LilyWangL — APPROVED** [review]

File: ports/glib/portfile.cmake [hunk 1]
```diff
@@ -50,9 +50,8 @@ vcpkg_install_cmake()
...
 
 vcpkg_copy_pdbs()
-vcpkg_copy_tool_dependencies(${CURRENT_PACKAGES_DIR}/tools/glib)
+vcpkg_copy_tool_dependencies(${CURRENT_PACKAGES_DIR}/tools/${PORT})
 
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
 
-file(COPY ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/glib)
-file(RENAME ${CURRENT_PACKAGES_DIR}/share/glib/COPYING ${CURRENT_PACKAGES_DIR}/share/glib/copyright)
+file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```
Comment (strega-nil) [inline ports/glib/portfile.cmake:57]: Could you make sure there's a newline at the end of this file?

**Comment by strega-nil** [comment]
Other than that, looks good to me :)

**Comment by strega-nil** [comment]
Any idea what's going on with libnice and cairo?

**Comment by NancyLi1013** [comment]
@strega-nil 
It seems all checks have passed on CI.

**Comment by strega-nil** [comment]
Cool, thanks @NancyLi1013 :)

**Pull request merged**